search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/prompt/PromptUtil.java

```diff
@@ -87,6 +92,48 @@ enum ChatRole {
         }
     }
 
+    public static String buildSingleStringPrompt(
```

Comment:
Is this also bedrock specific? or it can work on other LLMs as well? if bedrock specific for now, let's add a comment.

Reply:
This should work for any LLM - this just crunches the entire input into a single string, so any connector that can take a single string as input should be able to cooperate. 
Not compatible with trying to do more fancy things like with openai, passing a list of messages with roles and stuff.

Reply:
It doesn't have to be.  I've only dealt with two APIs (OpenAI and Bedrock/Anthropic).  It might just be that we have a "single string prompt" vs a "structured prompt".  I am not sure.  Please see the comment below.  I have some proposals, but wanted to discuss with the community about that.  At the moment, I think we can do one-offs for two or three providers which will cover more than 80% of the use cases.

Reply:
I'm wondering, currently, Bedrock supports multiple LLMs. What if buildSingleStringPrompt doesn't quite support Bedrock/Claude? How are we going to deal with this? Although bedrock is Llm.ModelProvider but what if claude is acting differently than Anthropic?

Reply:
As long as we haven't thoroughly tested with other Bedrock models, I feel that we should explicitly mention that currently, this feature is compatible with OpenAI and Bedrock/Anthropic.

Similarly, I would suggest adding comments accordingly, as we don't have enough time to test with other LLMs.

Reply:
I do have comments to that effect at least in two places in the code.

Reply:
I have a doc issue for 2.11.  I can see if they can still turn around and add this disclaimer, but I can't guarantee they will be able to make it.

Reply:
Doc issue - https://github.com/opensearch-project/documentation-website/issues/5125